### PR TITLE
#725: PR 3 / 4: Make enum values work with @Description annotation

### PR DESCRIPTION
**Pull request by PrimosK**
## Issue 

https://github.com/langchain4j/langchain4j/issues/725

PR 3 / 4: Makes it possible to use @Description on Enum values

Side notes:
1. This PR is dependant on PR [1 / 4.](https://github.com/langchain4j/langchain4j/pull/1389)
2. List<Enum> is still not supported - this will be tackled in [PR 4 / 4](https://github.com/langchain4j/langchain4j/pull/1410)

## Change

Made it possible to annotate Enum values with @Description.

This is how prompt looks like for enums without descriptions:
```
Analyze sentiment of:
|||This LaptopPro X15 is wicked fast and that 4K screen is a dream.|||
You must answer strictly with one of these enums:
POSITIVE
NEUTRAL
NEGATIVE
```

This is how prompt looks like for enums with descriptions:
```
Analyze weather forecast for:
|||It will be cloudy and mostly rainy. No more rain early in the day but the sky remains overcast. Afternoon it is mostly cloudy. The sun will not be visible. The forecast has a moderate, 40% chance of Precipitation. Temperatures peaking at 17 °C.|||
You must answer strictly with one of these enums:
SUNNY - A clear day with bright sunlight and few or no clouds
CLOUDY - The sky is covered with clouds with no rain, often creating a gray and overcast appearance
RAINY - Precipitation in the form of rain, with cloudy skies and wet conditions
SNOWY - Snowfall occurs, covering the ground in white and creating cold, wintry conditions
```

Covered everything with unit and integrationt tests.

Tested mainly with ˙gpt-3.5-turbo˙.

## General checklist

- [X] There are no breaking changes
- [X] I have added unit and integration tests for my change
- [X] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [X] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

### CODE REVIEW

File: langchain4j/src/main/java/dev/langchain4j/model/output/EnumOutputParser.java [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+import dev.langchain4j.model.output.structured.Description;
+
+import java.lang.reflect.Field;
+import java.util.Locale;
+import java.util.Optional;
 
 @SuppressWarnings("rawtypes")
 public class EnumOutputParser implements OutputParser<Enum> {
 
+    private static final boolean PRETTY = false;
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/output/EnumOutputParser.java]: is this required?

**Review by ghost — COMMENTED** [review]
@riferrei looks good!

**Comment by langchain4j** [comment]
@PrimosK please resolve the conflict and we can merge

File: langchain4j/src/main/java/dev/langchain4j/model/output/EnumOutputParser.java (hunk 1 above)
Comment (PrimosK) [reply langchain4j/src/main/java/dev/langchain4j/model/output/EnumOutputParser.java]: It's indeed nood needed... Obsolete artifact I've used for experimenting.

**Comment by PrimosK** [comment]
@langchain4j, conflicts resolved ...

**Comment by PrimosK** [comment]
@langchain4j, conflicts resolved ...

**Review by ghost — APPROVED** [review]
@PrimosK thank you!

**Pull request merged**